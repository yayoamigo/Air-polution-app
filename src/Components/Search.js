import React, { useRef } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { setSearch } from '../Redux/ducks/slices';

const Search = () => {
  const search = useSelector((state) => state.countries.search);
  const dispatch = useDispatch();
  const Input = useRef('');
  const handleSearch = () => {
    dispatch(setSearch(Input.current.value));
  };

  return (
    <div className="searchdiv">
      <input
        className="input"
        ref={Input}
        type="text"
        placeholder="search..."
        onChange={handleSearch}
        value={search}
      />
    </div>
  );
};

export default Search;
