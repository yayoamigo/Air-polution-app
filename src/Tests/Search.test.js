import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Search from '../Components/Search';
import store from '../Redux/configureStore';

it('render', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Search />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});