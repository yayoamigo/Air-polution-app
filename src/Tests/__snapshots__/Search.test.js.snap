// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<div
  className="searchdiv"
>
  <input
    className="input"
    onChange={[Function]}
    placeholder="search..."
    type="text"
    value=""
  />
</div>
`;
